src/test/java/com/google/cloud/tools/maven/it/GenRepoInfoFileMojoIntegrationTest.java

```diff
@@ -0,0 +1,39 @@
+package com.google.cloud.tools.maven.it;
+
+import com.google.cloud.tools.appengine.cloudsdk.internal.process.ProcessRunnerException;
+import com.google.cloud.tools.maven.it.verifier.FlexibleVerifier;
+import com.google.cloud.tools.maven.it.verifier.StandardVerifier;
+
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ * {@link com.google.cloud.tools.maven.GenRepoInfoFileMojo} integration tests.
+ */
+public class GenRepoInfoFileMojoIntegrationTest extends AbstractMojoIntegrationTest {
+  @Test
+  public void testGenerate() throws IOException, VerificationException {
+    Verifier verifier = new StandardVerifier("testGenRepoInfoFile");
+
+    verifier.executeGoal("appengine:genRepoInfoFile");
+    verifier.assertFilePresent("target/appengine-staging/WEB-INF/classes/source-context.json");
+  }
+
+  /**
+   * Ensures that the genRepoInfoFile goal is automatically called for a flexible project
+   * deployment.
+   */
+  @Test
+  public void testGenerateFlex() throws VerificationException, IOException, ProcessRunnerException {
+    Verifier verifier = new FlexibleVerifier("testGenRepoInfoFile_flex");
+
+    verifier.executeGoal("appengine:deploy");
```

Comment:
Added a test to check if goal is ran with package.

I would actually prefer to keep the deploy test, as source contexts on flex deployments were a big motivator for doing this. In this instance, I want to ensure that for a project with the executions bit in its pom.xml, the goal is automatically ran and the files are correctly added.